### PR TITLE
Add Github Actions

### PR DESCRIPTION
**Pull request by victor77dev**
### Description
- Add Dev lab & auto build envs

### QA
- [ ] Chrome

### Issuer
- https://github.com/Tremendous-Tiny-Team/ttt-web/issues/16

### TODO
- [ ] any follow-up here that is out of scope

### CODE REVIEW

**Comment by kavimaluskam** [comment]
@victor77dev I remember orginally the github action allow multiple node.js version to be tested in ci. Do you think it's necessary?

**Comment by victor77dev** [comment]
> @victor77dev I remember orginally the github action allow multiple node.js version to be tested in ci. Do you think it's necessary?

@kavimaluskam I don't think so. I think we are good to just support 1 node.js version.

File: package.json [hunk 1]
```diff
@@ -35,8 +35,8 @@
     "start": "npm run develop",
     "serve": "gatsby serve",
     "clean": "gatsby clean",
-    "test": "yarn run test:type && yarn run test:dumb",
-    "test:unit": "echo \"Write tests! -> https://gatsby.dev/unit-testing\" && exit 1",
+    "test": "yarn run test:type && yarn run test:unit",
```
Comment (victor77dev) [inline package.json:38]: @kavimaluskam Do you think I should put `npm` here? Is `yarn` here ok?

Comment (kavimaluskam) [reply package.json:38]: I personally think `yarn` is ok. 
Randomly searched some approach on internet as well:
<https://stackoverflow.com/questions/51768743/how-to-detect-that-the-script-is-running-with-npm-or-yarn>

Comment (victor77dev) [reply package.json:38]: Cool, but let's leave it as it is now. I prefer not install a lib to support cross platform for only this usage.
Using `yarn` looks good to me.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      ready_for_review,
+      reopened,
+      synchronize,
+    ]
+    branches:
+      - main
+      - release**
+      - production**
+      - hotfix**
+      - lab**
```
Comment (victor77dev) [inline .github/workflows/test.yml:17]: @kavimaluskam These are the branch which will run the test

Comment (kavimaluskam) [reply .github/workflows/test.yml:17]: Yes as pre-discussed today.
Will do review tmr.

**Comment by victor77dev** [comment]
The PR is ready to review and merge, i tried to trigger with "ready to review" & label trigger & cache `yarn install` with different `yarn.lock`.

File: .github/workflows/test.yml (hunk 2 above)
Comment (victor77dev) [reply .github/workflows/test.yml:17]: Take your time😉

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+      run: echo "::set-output name=dir::$(yarn cache dir)"
+
+    - uses: actions/cache@v2
+      id: yarn-cache
+      with:
+        path: |
+          ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          **node_modules
+        key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+        restore-keys: |
```
Comment (victor77dev) [inline .github/workflows/test.yml:49]: I just noticed I shouldn't set `restore-keys` with hash
Reference: https://github.community/t/understanding-cache-action-keys/17503/2

Comment (kavimaluskam) [reply .github/workflows/test.yml:49]: so `restore-key` is like a fall-back / backup plans for the cache key?

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+      - release**
+      - production**
+      - hotfix**
+      - lab-**
+
+env:
+  NODE_VERSION: '14.x'
+
+jobs:
+  Type:
```
Comment (kavimaluskam) [inline .github/workflows/test.yml:23]: the job name `Type` here looks a bit confusing (despite of the workflow name is test), shall we name it `Type-Test`, `Lint-Test`, or something?

**Review by kavimaluskam — CHANGES_REQUESTED** [review]
Looks good to me except the naming part.
And feels the build / deploy path is the real beast to deal with XD

**Review by kavimaluskam — APPROVED** [review]

**Pull request merged**